eduHub/src/main/kotlin/com/arsiu/eduhub/kafka/consumer/AssignmentKafkaConsumer.kt

```diff
@@ -0,0 +1,34 @@
+package com.arsiu.eduhub.kafka.consumer
+
+import com.arsiu.eduhub.v2.assignmentsvc.NatsSubject
+import com.arsiu.eduhub.v2.assignmentsvc.commonmodels.assignment.AssignmentProto
+import io.nats.client.Connection
+import jakarta.annotation.PostConstruct
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Component
+import reactor.kafka.receiver.KafkaReceiver
+
+@Component
+class AssignmentKafkaConsumer(
+    private val assignmentKafkaReceiver: KafkaReceiver<String, AssignmentProto>,
+    private val natsConnection: Connection
+) {
+
+    private val logger: Logger = LoggerFactory.getLogger(AssignmentKafkaConsumer::class.java)
+    private val subject = NatsSubject.ASSIGNMENT_UPDATE_BUS
+
+    @PostConstruct
+    fun listen() {
+        assignmentKafkaReceiver.receiveAutoAck()
+            .concatMap { fluxRecord ->
+                fluxRecord.map { record ->
+                    natsConnection.publish(subject, record.value().toByteArray())
+                    record
+                }
+            }.doOnError { e ->
```

Comment:
```suggestion
            .flatMap { 
                    natsConnection.publish(subject, it.value().toByteArray())
                    .thenReturn(it)
                }
            }.doOnError { e ->
```
can it be rewritten like in my suggestions?

Reply:
can be rewritten using flatMap twice
because receiveAutoAck returns Flux<Flux<ConsumerRecord>>

```
.flatMap { fluxRecord ->
            fluxRecord.flatMap { record ->
                publishToNats(record)
                    .thenReturn(record)
            }
        }
        .doOnError { e ->
```

So at first I think its better looking when use concatMap and map
but now  I see that smells.

will be rewrite to flatMap twice

Reply:
or that even worse ?

Reply:
It's ok, I didn't understand why you needed concatMap